osu.Game/Beatmaps/BeatmapDifficultyManager.cs

```diff
@@ -220,6 +237,14 @@ private bool tryGetExisting(BeatmapInfo beatmapInfo, RulesetInfo rulesetInfo, IE
             return difficultyCache.TryGetValue(key, out existingDifficulty);
         }
 
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            cancelTrackedBindableUpdate();
+            updateScheduler.Dispose();
```

Comment:
Probably safer to null-guard this as this can potentially be called from the finaliser (if `isDisposing == false`) and the object is not guaranteed to be fully initialised in that case.

```suggestion
            updateScheduler?.Dispose();
```